sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -696,6 +696,37 @@ case class ResolvedStar(expressions: Seq[NamedExpression]) extends Star with Une
   override def toString: String = expressions.mkString("ResolvedStar(", ", ", ")")
 }
 
+/**
+ * Represents all input attributes to a given relational operator.
+ * This is used in Spark Connect dataframe, for example:
+ *    df1 = spark.createDataFrame([{"id": 1}])
+ *    df2 = spark.createDataFrame([{"id": 1, "val": "v"}])
+ *    df1.join(df2, "id").select(df1["*"])
+ * @param planId the plan id of target node.
+ */
+case class UnresolvedDataFrameStar(planId: Long) extends Star with Unevaluable {
+  override def expand(input: LogicalPlan, resolver: Resolver): Seq[NamedExpression] = {
+    val resolved = resolveDFStarRecursively(planId, input)
+    resolved.map(_.expand(input, resolver)).getOrElse(
+      throw QueryCompilationErrors.cannotResolveStar(this)
+    )
+  }
+
+  private def resolveDFStarRecursively(
+    id: Long,
+    p: LogicalPlan): Option[ResolvedStar] = {
+    val resolved = if (p.getTagValue(LogicalPlan.PLAN_ID_TAG).contains(id)) {
+      Some(ResolvedStar(p.output))
+    } else {
+      p.children.iterator.map(resolveDFStarRecursively(id, _))
+        .foldLeft(Option.empty[ResolvedStar]) {
```

Comment:
don't add an ambiguous detection for now, since in vanilla Spark,

```
In [7]: df1 = spark.createDataFrame([{"id": 1}])

In [8]: df1.join(df1)
Out[8]: DataFrame[id: bigint, id: bigint]

In [9]: df1.join(df1).select(df1["id"])
...
AnalysisException: Column id#0L are ambiguous. It's probably because you joined several Datasets together, and some of these Datasets are the same. 

In [10]: df1.join(df1).select(df1["*"])
Out[10]: DataFrame[id: bigint]
```

Reply:
It's probably a bug in vanilla spark...

Reply:
yeah, let me fail it in spark connect anyway